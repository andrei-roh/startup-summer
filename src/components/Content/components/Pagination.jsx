import React, {useState} from 'react';
import Pagination from '@material-ui/lab/Pagination';
import Typography from '@material-ui/core/Typography';
import {
  StyledCard,
  PaginationLink,
  RightBlock,
  BottomBlock
} from './style';

const PaginationBlock = ({ userRepositoryInfo, public_repos }) => {
  const [page, setPage] = useState(1);
  const handleChange = (event, value) => {
    setPage(value);
  };
  let stepsArray = [0];
  let step = 4
  for (let i = 0; i < (public_repos / 4); i += 1) {
  	stepsArray.push(step);
    step += 4
  }
  let paginateUserRepositoryInfo = (pageNumber) => {
    return userRepositoryInfo.slice(stepsArray[pageNumber - 1], stepsArray[pageNumber])
  };
  let getNecessaryRepositoryInfo = paginateUserRepositoryInfo(page).map(element =>
    <StyledCard>
      <Typography variant="body1">
        <PaginationLink href={element.html_url} target="_blanc">{element.name}</PaginationLink>
      </Typography>
      <Typography variant="body2">{element.description}</Typography>
    </StyledCard>
  );
  let getPaginationCount = Math.floor(public_repos / 4);

  return (
    <RightBlock>
      <div>{getNecessaryRepositoryInfo}</div>
      <BottomBlock>
        <Typography variant="body2" style={{ color: "#808080" }}>
          {stepsArray[page - 1] === 0 ? 1 : stepsArray[page - 1]}-{stepsArray[page]} of {public_repos} items
        </Typography>
        <Pagination
          count={getPaginationCount}
          page={page}
          onChange={handleChange}
          shape="rounded"
          color="primary"
        />
      </BottomBlock>
    </RightBlock>
  );
}

export default PaginationBlock;
