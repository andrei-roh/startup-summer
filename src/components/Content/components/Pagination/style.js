import styled from 'styled-components';
import Card from '@material-ui/core/Card';

export const StyledCard = styled(Card)`
  width: 100%;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 16px 24px 16px 24px;
  box-sizing: border-box;
  margin-top: 12px;
  margin-bottom: 12px;
  background: #fff;

  @media (max-width: 799px) {
    width: 90%;
    margin-right: auto;
    margin-left: auto;
  }
`;

export const PaginationLink = styled.a`
  text-decoration: none;
  color: #0064eb;
  background: #fff;
`;

export const RightBlock = styled.div`
  display: flex;
  flex-direction: column;
  height: 70vh;
  justify-content: space-between;
  width: 50vw;
  align-self: flex-end;
  background: #e5e5e5;

  @media (max-width: 799px) {
    flex-direction: column;
    align-items: center;
    justify-content: center;
    width: 100%;
    height: 100%;
    align-self: flex-start;
  }
`;

export const BottomBlock = styled.div`
  display: flex;
  align-self: flex-end;
  flex-direction: row;
  align-items: center;

  @media (max-width: 799px) {
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    width: 95vw;
    align-self: flex-start;
    margin-bottom: 15px;
  }
`;
