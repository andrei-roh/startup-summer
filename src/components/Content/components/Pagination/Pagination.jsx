import React, { useState } from 'react';
import Pagination from '@material-ui/lab/Pagination';
import Typography from '@material-ui/core/Typography';
import { StyledCard, PaginationLink, RightBlock, BottomBlock } from './style';

const PaginationBlock = ({ userRepositoryInfo, public_repos }) => {
  const [page, setPage] = useState(1);
  const handleChange = (event, value) => {
    setPage(value);
  };

  let stepsArray = [0];
  let step = 4;
  for (let i = 0; i < public_repos / 4; i += 1) {
    stepsArray.push(step);
    step += 4;
  }
  let paginateUserRepositoryInfo = (pageNumber) => {
    return userRepositoryInfo
      .slice(0, public_repos)
      .slice(stepsArray[pageNumber - 1], stepsArray[pageNumber]);
  };
  let getNecessaryRepositoryInfo = paginateUserRepositoryInfo(page).map(
    (element) => (
      <StyledCard>
        <Typography variant="body1">
          <PaginationLink href={element.html_url} target="_blanc">
            {element.name}
          </PaginationLink>
        </Typography>
        <Typography variant="body2">{element.description}</Typography>
      </StyledCard>
    )
  );

  let getPaginationCount = Math.ceil(public_repos / 4);
  const firstNumberItems = () => {
    if (stepsArray[page - 1] === 0) {
      return 1;
    }
    return stepsArray[page - 1];
  };
  const lastNumberItems = () => {
    if (page === getPaginationCount) {
      return public_repos;
    }
    return stepsArray[page];
  };

  return (
    <RightBlock>
      <Typography variant="h5" style={{ fontWeight: 'bold' }}>
        Repositories ({public_repos})
      </Typography>
      <div>{getNecessaryRepositoryInfo}</div>
      <BottomBlock>
        <Typography variant="body2" style={{ color: '#808080' }}>
          {firstNumberItems()}-{lastNumberItems()} of {public_repos} items
        </Typography>
        <Pagination
          count={getPaginationCount}
          page={page}
          onChange={handleChange}
          shape="rounded"
          color="primary"
        />
      </BottomBlock>
    </RightBlock>
  );
};

export default PaginationBlock;
