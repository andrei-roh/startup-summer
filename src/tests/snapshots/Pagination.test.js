import React from 'react';
import { shallow } from 'enzyme';
import Pagination from '../../components/Content/components/Pagination/Pagination';

const userInfo = [
  {
    id: 295691757,
    node_id: 'MDEwOlJlcG9zaXRvcnkyOTU2OTE3NTc=',
    name: 'basic-js',
    full_name: 'andrei-roh/basic-js',
    private: false,
    owner: {
      login: 'andrei-roh',
      id: 65450338,
      node_id: 'MDQ6VXNlcjY1NDUwMzM4',
      avatar_url: 'https://avatars.githubusercontent.com/u/65450338?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/andrei-roh',
      html_url: 'https://github.com/andrei-roh',
      followers_url: 'https://api.github.com/users/andrei-roh/followers',
      following_url:
        'https://api.github.com/users/andrei-roh/following{/other_user}',
      gists_url: 'https://api.github.com/users/andrei-roh/gists{/gist_id}',
      starred_url:
        'https://api.github.com/users/andrei-roh/starred{/owner}{/repo}',
      subscriptions_url:
        'https://api.github.com/users/andrei-roh/subscriptions',
      organizations_url: 'https://api.github.com/users/andrei-roh/orgs',
      repos_url: 'https://api.github.com/users/andrei-roh/repos',
      events_url: 'https://api.github.com/users/andrei-roh/events{/privacy}',
      received_events_url:
        'https://api.github.com/users/andrei-roh/received_events',
      type: 'User',
      site_admin: false,
    },
    html_url: 'https://github.com/andrei-roh/basic-js',
    description: 'Basic JavaScript tasks for RSSchool',
    fork: true,
    url: 'https://api.github.com/repos/andrei-roh/basic-js',
    forks_url: 'https://api.github.com/repos/andrei-roh/basic-js/forks',
    keys_url: 'https://api.github.com/repos/andrei-roh/basic-js/keys{/key_id}',
    collaborators_url:
      'https://api.github.com/repos/andrei-roh/basic-js/collaborators{/collaborator}',
    teams_url: 'https://api.github.com/repos/andrei-roh/basic-js/teams',
    hooks_url: 'https://api.github.com/repos/andrei-roh/basic-js/hooks',
    issue_events_url:
      'https://api.github.com/repos/andrei-roh/basic-js/issues/events{/number}',
    events_url: 'https://api.github.com/repos/andrei-roh/basic-js/events',
    assignees_url:
      'https://api.github.com/repos/andrei-roh/basic-js/assignees{/user}',
    branches_url:
      'https://api.github.com/repos/andrei-roh/basic-js/branches{/branch}',
    tags_url: 'https://api.github.com/repos/andrei-roh/basic-js/tags',
    blobs_url:
      'https://api.github.com/repos/andrei-roh/basic-js/git/blobs{/sha}',
    git_tags_url:
      'https://api.github.com/repos/andrei-roh/basic-js/git/tags{/sha}',
    git_refs_url:
      'https://api.github.com/repos/andrei-roh/basic-js/git/refs{/sha}',
    trees_url:
      'https://api.github.com/repos/andrei-roh/basic-js/git/trees{/sha}',
    statuses_url:
      'https://api.github.com/repos/andrei-roh/basic-js/statuses/{sha}',
    languages_url: 'https://api.github.com/repos/andrei-roh/basic-js/languages',
    stargazers_url:
      'https://api.github.com/repos/andrei-roh/basic-js/stargazers',
    contributors_url:
      'https://api.github.com/repos/andrei-roh/basic-js/contributors',
    subscribers_url:
      'https://api.github.com/repos/andrei-roh/basic-js/subscribers',
    subscription_url:
      'https://api.github.com/repos/andrei-roh/basic-js/subscription',
    commits_url:
      'https://api.github.com/repos/andrei-roh/basic-js/commits{/sha}',
    git_commits_url:
      'https://api.github.com/repos/andrei-roh/basic-js/git/commits{/sha}',
    comments_url:
      'https://api.github.com/repos/andrei-roh/basic-js/comments{/number}',
    issue_comment_url:
      'https://api.github.com/repos/andrei-roh/basic-js/issues/comments{/number}',
    contents_url:
      'https://api.github.com/repos/andrei-roh/basic-js/contents/{+path}',
    compare_url:
      'https://api.github.com/repos/andrei-roh/basic-js/compare/{base}...{head}',
    merges_url: 'https://api.github.com/repos/andrei-roh/basic-js/merges',
    archive_url:
      'https://api.github.com/repos/andrei-roh/basic-js/{archive_format}{/ref}',
    downloads_url: 'https://api.github.com/repos/andrei-roh/basic-js/downloads',
    issues_url:
      'https://api.github.com/repos/andrei-roh/basic-js/issues{/number}',
    pulls_url:
      'https://api.github.com/repos/andrei-roh/basic-js/pulls{/number}',
    milestones_url:
      'https://api.github.com/repos/andrei-roh/basic-js/milestones{/number}',
    notifications_url:
      'https://api.github.com/repos/andrei-roh/basic-js/notifications{?since,all,participating}',
    labels_url:
      'https://api.github.com/repos/andrei-roh/basic-js/labels{/name}',
    releases_url:
      'https://api.github.com/repos/andrei-roh/basic-js/releases{/id}',
    deployments_url:
      'https://api.github.com/repos/andrei-roh/basic-js/deployments',
    created_at: '2020-09-15T10:24:44Z',
    updated_at: '2021-07-01T17:01:03Z',
    pushed_at: '2020-10-08T06:31:55Z',
    git_url: 'git://github.com/andrei-roh/basic-js.git',
    ssh_url: 'git@github.com:andrei-roh/basic-js.git',
    clone_url: 'https://github.com/andrei-roh/basic-js.git',
    svn_url: 'https://github.com/andrei-roh/basic-js',
    homepage: '',
    size: 185,
    stargazers_count: 0,
    watchers_count: 0,
    language: 'JavaScript',
    has_issues: false,
    has_projects: true,
    has_downloads: true,
    has_wiki: true,
    has_pages: false,
    forks_count: 0,
    mirror_url: null,
    archived: false,
    disabled: false,
    open_issues_count: 0,
    license: {
      key: 'mit',
      name: 'MIT License',
      spdx_id: 'MIT',
      url: 'https://api.github.com/licenses/mit',
      node_id: 'MDc6TGljZW5zZTEz',
    },
    allow_forking: true,
    visibility: 'public',
    forks: 0,
    open_issues: 0,
    watchers: 0,
    default_branch: 'master',
  },
];

describe('<Pagination />', () => {
  const initialProps = {
    userRepositoryInfo: userInfo,
  };
  const pagination = shallow(<Pagination {...initialProps} />);
  it('should match the snapshot', () => {
    expect(pagination.html()).toMatchSnapshot();
  });
});
