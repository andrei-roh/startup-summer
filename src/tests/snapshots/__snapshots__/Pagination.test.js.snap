// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination /> should match the snapshot 1`] = `"<div class=\\"sc-dlnjwi bIUCZN\\"><h5 class=\\"MuiTypography-root MuiTypography-h5\\" style=\\"font-weight:bold\\">Repositories ()</h5><div><div class=\\"MuiPaper-root MuiCard-root sc-bdnxRM hsYeIT MuiPaper-elevation1 MuiPaper-rounded\\"><p class=\\"MuiTypography-root MuiTypography-body1\\"><a href=\\"https://github.com/andrei-roh/basic-js\\" target=\\"_blanc\\" class=\\"sc-gtsrHT kgWjQH\\">basic-js</a></p><p class=\\"MuiTypography-root MuiTypography-body2\\">Basic JavaScript tasks for RSSchool</p></div></div><div class=\\"sc-hKFxyN dFQwXI\\"><p class=\\"MuiTypography-root MuiTypography-body2\\" style=\\"color:#808080\\">1- of  items</p><nav aria-label=\\"pagination navigation\\" class=\\"MuiPagination-root\\"><ul class=\\"MuiPagination-ul\\"><li><button class=\\"MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-rounded MuiPaginationItem-textPrimary Mui-disabled Mui-disabled\\" tabindex=\\"-1\\" type=\\"button\\" disabled=\\"\\" aria-label=\\"Go to previous page\\"><svg class=\\"MuiSvgIcon-root MuiPaginationItem-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z\\"></path></svg></button></li><li><button class=\\"MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-rounded MuiPaginationItem-textPrimary\\" tabindex=\\"0\\" type=\\"button\\" aria-label=\\"Go to next page\\"><svg class=\\"MuiSvgIcon-root MuiPaginationItem-icon\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\"><path d=\\"M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z\\"></path></svg></button></li></ul></nav></div></div>"`;
